Poker.BE/Poker.BE.Domain.Tests/Game/RoundTests.cs

```diff
@@ -0,0 +1,100 @@
+            Exception expectedException2 = new Exception();
+
+            //Act
+            round.PlayMove(Round.Move.bet, 50);
+            var res1 = round.CurrentPlayer == player2 && pot.PlayersClaimPot.Contains(player1) && round.LiveBets[player1] == 50
+                        && pot.AmountToClaim == 50 && pot.Value == 50 && round.LastRaise == 50 && round.TotalRaise == 50;
+
+            round.PlayMove(Round.Move.call, 0);
+            var res2 = round.CurrentPlayer == player3 && pot.PlayersClaimPot.Contains(player2) && round.LiveBets[player2] == 50
+                        && pot.AmountToClaim == 50 && pot.Value == 100 && round.LastRaise == 50 && round.TotalRaise == 50;
+
+            try
+            {
+                round.PlayMove(Round.Move.raise, 20);
+            }
+            catch(ArgumentException e)
+            {
+                expectedException1 = e;
+            }
+
+            round.PlayMove(Round.Move.raise, 70);
+            var res3 = round.CurrentPlayer == player1 && pot.PlayersClaimPot.Contains(player3) && round.LiveBets[player3] == 120
+                        && pot.AmountToClaim == 120 && pot.Value == 220 && round.LastRaise == 70 && round.TotalRaise == 120;
+
+            round.PlayMove(Round.Move.allin, 0);
+            var res4 = round.CurrentPlayer == player2 && pot.PlayersClaimPot.Contains(player1) && round.LiveBets[player1] == 500
+                        && pot.AmountToClaim == 500 && pot.Value == 670 && round.LastRaise == 380 && round.TotalRaise == 500
+                        && pot.PartialPot != null && pot.PartialPot.AmountToClaim == 0 && pot.PartialPot.Value == 0;
+
+            round.PlayMove(Round.Move.allin, 0);
+            var res5 = round.CurrentPlayer == player3 && round.CurrentPot.PlayersClaimPot.Contains(player1) && round.CurrentPot.PlayersClaimPot.Contains(player2)
+                        && !round.CurrentPot.PartialPot.PlayersClaimPot.Contains(player2) && round.CurrentPot.PartialPot.PlayersClaimPot.Contains(player1)
+                        && round.LiveBets[player2] == 350 && round.CurrentPot.AmountToClaim == 350 && round.CurrentPot.Value == 820  
+                        && round.CurrentPot.PartialPot.Value == 150 && round.CurrentPot.PartialPot.AmountToClaim == 150;
+
+            try
+            {
+                round.PlayMove(Round.Move.allin, 0);
+            }
+            catch (IOException e)
+            {
+                expectedException2 = e;
+            }
+
+            //round.PlayMove(Round.Move.fold, 0);
+            var res6 = !round.CurrentPot.PlayersClaimPot.Contains(player3) && !round.CurrentPot.PartialPot.PlayersClaimPot.Contains(player3)
+                        && round.LiveBets[player3] == 120 && round.CurrentPot.AmountToClaim == 350 && round.CurrentPot.Value == 820
+                        && round.CurrentPot.PartialPot.Value == 150 && round.CurrentPot.PartialPot.AmountToClaim == 150;
+
+            //Assert
```

Comment:
basically you're right, but in this specific test i wanted to simulate the flow of the round, using a number of players who play a different play move each time and depend on the previous player's play move.